### DISCUSSION TITLE
WiX 4 HarvestDirectory generating list of files but not included in the msi

### DISCUSSION THREAD

**Discussion by bickle** [wixtoolset/issues#8017] [Questions]
I'm using WiX toolset v4 and using HarvestDirectory. It is generating the temp file with the appropriate list of files, but they are not being included in the msi and I receive an error message of "WIX1076: ICE71: The Media table has no entries". I thought I had the component groups and directories set, but I'm unable to see what I am missing or did incorrectly.

In the wixproj file I have:
```
    <ItemGroup>
        <HarvestDirectory Include="$(IntermediateOutputPath)publish\WinFormApp" >
            <ComponentGroupName>HarvestedComponents</ComponentGroupName>
            <DirectoryRefId>INSTALLFOLDER</DirectoryRefId>
            <SuppressRootDirectory>true</SuppressRootDirectory>
            <SuppressRegistry>true</SuppressRegistry>
        </HarvestDirectory>
    </ItemGroup>
  <ItemGroup>
    <ProjectReference Include="..\src\WinFormApp\WinFormApp.csproj" Publish="true" />
  </ItemGroup>
```

In my Product.wxs file I have:
```
    <Fragment>
	    <StandardDirectory Id="ProgramFilesFolder">
            <Directory Id="CompanyFolder" Name="Default Company Name">
			    <Directory Id="INSTALLFOLDER" Name="WinForm App" />
		    </Directory>
        </StandardDirectory>
    </Fragment>
	
    <Fragment>
        <ComponentGroup Id="HarvestedComponents" Directory="INSTALLFOLDER">
            <ComponentGroupRef Id="HarvestedComponents"/>
        </ComponentGroup>
    </Fragment>
```
I've been following along with the  "Harvesting the hard way with Heat in WiX v4" video, but I'm not seeing what I am missing.

Thanks

**Comment by robmen** [comment]
You're not referencing `ComponentGroup` in the `Fragment`. I'd recommend watching the earlier Deployment Dojo shows to get a few more fundamentals under your belt. Jumping in at episode 52 means you missed a lot of context.

Or read the [WiX v4 tutorial by FireGiant](https://www.firegiant.com/docs/wix/tutorial/). It'll teach you the basics you are lacking.
